index.d.ts

```diff
@@ -1,9 +1,14 @@
-import { SnakeCase } from "type-fest";
+import { SnakeCase, JsonPrimitive } from "type-fest";
 import { Options as SnakeCaseOptions } from "snake-case";
 
 // eslint-disable-next-line @typescript-eslint/ban-types
 type EmptyTuple = [];
 
+type JsonObject = { [Key in string]: JsonValue };
+type JsonObjectOptional = JsonObject | undefined;
+type JsonValue = JsonPrimitive | object | JsonObject | JsonArray | undefined;
+type JsonArray = JsonValue[] | readonly JsonValue[];
```

Comment:
I referred to type-fest for the following type definition, but would `Record<string, unknown>` be better?
I have confirmed that the type is as expected (test is All Green) even with `Record<string, unknown>`.

```ts
type JsonObject = { [Key in string]: JsonValue };
type JsonObjectOptional = JsonObject | undefined;
type JsonValue = JsonPrimitive | object | JsonObject | JsonArray | undefined;
type JsonArray = JsonValue[] | readonly JsonValue[];
```

Reply:
Yeah, seems like it. I'm not an expert on TypeScript but it seems like assuming that all values are `JsonValue` is not necessarily safe.

Reply:
Thank you. I push fixes to `Record<string, unknown>`.